CDS/src/org/icpc/tools/cds/CDSIdentityStore.java

```diff
@@ -0,0 +1,26 @@
+package org.icpc.tools.cds;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.security.enterprise.credential.UsernamePasswordCredential;
+import javax.security.enterprise.identitystore.CredentialValidationResult;
+import javax.security.enterprise.identitystore.IdentityStore;
+
+import org.icpc.tools.contest.model.IAccount;
+
+@ApplicationScoped
+public class CDSIdentityStore implements IdentityStore {
+	public CredentialValidationResult validate(UsernamePasswordCredential userCredential) {
```

Comment:
Ah so maybe we *could* do this for IP auth, if we add some magic here. Although I'm not sure if we really want it

Reply:
Agreed. One implication is that you wouldn't be able to log into that account from any other machine.